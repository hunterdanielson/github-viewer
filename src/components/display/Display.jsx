import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ name, followers, following, html_url }) => (
  <>
    <h1>{name}</h1>
    <p>Followers: {followers} / Following: {following}</p>
    <a href={html_url}>Link</a>
  </>
);

Display.propTypes = {
  name: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  html_url: PropTypes.string.isRequired
};

export default Display;
