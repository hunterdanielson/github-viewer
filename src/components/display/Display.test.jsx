import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(<Display name='hunter' followers={1} following={2} html_url='www.github.com/hunterdanielson' repos={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
