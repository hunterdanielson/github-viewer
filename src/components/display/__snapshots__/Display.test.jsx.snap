// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<Fragment>
  <h1>
    hunter
  </h1>
  <p>
    Followers: 
    1
     / Following: 
    2
  </p>
  <a
    href="www.github.com/hunterdanielson"
  >
    Profile Link
  </a>
</Fragment>
`;
